### PR TITLE
backend: document_pid on Loan create

### LINKED ISSUE #85: loans: add the `document_pid` on loans created on items
Currently when you create a loan on an item the document_pid is not attached on the loan. We should enable this behaviour by automatically attach the document_pid of the item.

### PR DESCRIPTION
**Pull request by topless**
* resolves #85

### CODE REVIEW

File: invenio_circulation/api.py [hunk 1]
```diff
@@ -34,8 +34,9 @@ def create(cls, data, id_=None, **kwargs):
         """Create Loan record."""
         data["$schema"] = current_jsonschemas.path_to_url(cls._schema)
         ref_builder = current_app.config.get("CIRCULATION_ITEM_REF_BUILDER")
-        item_attached = data.get("item_pid") and data["item_pid"] != ""
-        if ref_builder and item_attached:
+        item_pid = data.get("item_pid")
```
Comment (ntarocco) [inline invenio_circulation/api.py:37]: I think this like was specifically checking for `None` or empty string `''`.
Are you sure we want to remove it?

Comment (topless) [reply invenio_circulation/api.py:37]: @ntarocco If I understood correctly,
the statement `if item_attached: ` will fail in both cases `None` or `''`, so I thought it was an extra unnecessary check.

If it doesn't break anything I don't see why not remove it.

File: invenio_circulation/api.py [hunk 2]
```diff
@@ -34,8 +34,9 @@ def create(cls, data, id_=None, **kwargs):
...
+        if ref_builder and item_pid:
+            data["document_pid"] = get_document_by_item_pid(item_pid)
```
Comment (equadon) [inline invenio_circulation/api.py]: Maybe `get_document_by_item_pid` should be renamed to `get_document_pid_by_item_pid`. What do you think?

Comment (topless) [reply invenio_circulation/api.py]: Well obviously since it returns the `document_pid`, nice catch ;)

File: invenio_circulation/api.py (hunk 1 above)
Comment (zzacharo) [reply invenio_circulation/api.py:37]: Do we still set `item_pid` to empty string in the UI when item is not attached? That's why we were checking explicitly here I think.

Comment (topless) [reply invenio_circulation/api.py:37]: @zzacharo Yes the `item_pid` and is an empty string if there is no item attached

**Review by zzacharo — APPROVED** [review]

**Pull request merged**